src/features/counterfactual/utils.ts

```diff
@@ -221,6 +193,30 @@ export const showSubmitNotification = (dispatch: AppDispatch, chain?: ChainInfo,
   )
 }
 
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+
+/**
+ * Calling getTransaction too fast sometimes fails because the txHash hasn't been
+ * picked up by any node yet so we should retry a few times with short delays to
+ * make sure the transaction really does/does not exist
+ * @param provider
+ * @param txHash
+ * @param maxAttempts
+ * @param delayMs
+ */
+async function retryGetTransaction(provider: Provider, txHash: string, maxAttempts = 3, delayMs = 1000) {
+  for (let attempt = 0; attempt < maxAttempts; attempt++) {
+    const txResponse = await provider.getTransaction(txHash)
+    if (txResponse !== null) {
+      return txResponse
+    }
+    if (attempt < maxAttempts - 1) {
+      await delay(delayMs)
+    }
+  }
+  throw new Error('Transaction not found')
+}
```

Comment:
We have the exponential-backoff package for that. Possible to use that?

Reply:
I have a local stash to replace that package with our own implementation so I would leave it as is for now since we will remove that dependency anyway.